frontend/src/lib/utils.ts

```diff
@@ -203,3 +205,33 @@ export function labelVisibilityProtoValueToEnum(
       return LabelVisibilityOptions.Visible
   }
 }
+
+export function removeURLParam(paramKey: string): void {
+  const currentURL = new URL(document.URL)
+  const urlParams = new URLSearchParams(currentURL.search)
+  urlParams.delete(paramKey)
+  const search =
+    urlParams.toString().length > 0 ? `?${urlParams.toString()}` : ""
+  const newURL =
+    currentURL.origin + currentURL.pathname + search + currentURL.hash
+  window.history.replaceState(null, document.title, newURL)
+}
+
+export function isURLParamSetToParamValue(
+  urlParamKey: string,
+  urlParamValue: string
+): boolean {
+  const urlParams = new URLSearchParams(new URL(document.URL).search)
+  return urlParams.get(urlParamKey) === urlParamValue
+}
+
+export function printAppWhenPrintURLParamIsSetCallback(): void {
```

Comment:
I'm not sure if utils are the right place for callbacks. Callback is something very specific and utils is a place for a general purpose stuff.

Reply:
Agree, I've moved the func to `App.tsx` -> `printWhenPrintURLParamIsSetCallback` 👍